use latest version SDK and go

gthao313 opened the pull request:
<!--
Tips:
- Please read the CONTRIBUTING document to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**
#52

**Description of changes:**

- Building process uses old version SDK version, update makefile to use latest SDK verison.
- Update go version

**Testing done:**
Building images

1. Building a bottlerocket update operator image.
```
[ec2-user@ip-172-31-38-250 bottlerocket-update-operator]$ docker image ls
REPOSITORY                                             TAG                        IMAGE ID       CREATED              SIZE
update-operator-arm64                                  v0.1.4-c451ec2e            eed93f8bd863   19 seconds ago       35MB
<none>                                                 <none>                     4c52d26e902a   30 seconds ago       4.26GB
<none>                                                 <none>                     fe2681a37757   57 seconds ago       209kB
```
2.Push image to ECR Public, and deploy to an aarch64 bottlerocket host. and Extract logs to check if it run successfully


```
kubectl apply -f ./update-operator.yaml

kubectl label node $(kubectl get nodes -o jsonpath='{.items[*].metadata.name}') bottlerocket.aws/updater-interface-version=2.0.0
```

```
kubectl get nodes -o json | jq -C -S '.items | map(.metadata|{(.name): (.annotations|to_entries|map(select(.key|startswith("bottlerock
et")))|from_entries)}) | add'
{
  "ip-192-168-67-242.us-west-2.compute.internal": {
    "bottlerocket.aws/action-active": "reboot-update",
    "bottlerocket.aws/action-state": "ready",
    "bottlerocket.aws/action-wanted": "reboot-update",
    "bottlerocket.aws/update-available": "true"
  }
}
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

srgothi92 commented on CHANGELOG.md:
```diff
@@ -33,7 +33,7 @@ kubectl label node --overwrite=true $(kubectl get nodes -o jsonpath='{.items[*].
 * Specify passing `-c` to `watch` in README instructions for monitoring node status ([#27])
 * Bump [bottlerocket-sdk](https://github.com/bottlerocket-os/bottlerocket-sdk) version to v0.10.1 for building the update operator's binaries. ([#21])
-* Bump the version of the golang image to 1.14.1 to match the Go toolchain version in the [bottlerocket-sdk](https://github.com/bottlerocket-os/bottlerocket-sdk). ([#31])
+* Bump the version of the golang image to 1.16.5 to match the Go toolchain version in the [bottlerocket-sdk](https://github.com/bottlerocket-os/bottlerocket-sdk). ([#31])
```
I don't think we should modify line in released version. You can add a Changelog for new version that we planned to release next and copy this line in that. You might also want to change the issue number.

gthao313 replied:
> Don't know anything about buildx plugin would let others review that part. I only reviewed the Go version change. Just confirming, Did you also repeated the testing after updating the Go version?

Yeah, I repeated the test and worked as expected. Also, I just repeated the test for my new changes and everything looks good. Thanks

srgothi92 reviewed (commented):
Don't know anything about `buildx plugin` would let others review that part. I only reviewed the Go version change. Just confirming, Did you also repeated the testing after updating the Go version?

gthao313 commented:
Push above remove change in `Changelog` according to @srgothi92 comment.

srgothi92 reviewed (approved):
Version change LGTM.

samuelkarp commented on Makefile:
```diff
@@ -22,7 +22,8 @@ SHORT_SHA = $(shell git describe --abbrev=8 --always --dirty='-dev' --exclude '*
 # BUILDSYS_SDK_IMAGE is the Bottlerocket SDK image used for license scanning.
-BUILDSYS_SDK_IMAGE ?= bottlerocket/sdk-$(UNAME_ARCH):v0.10.1
+BUILDSYS_SDK_VERSION=v0.22.0
+BUILDSYS_SDK_IMAGE = public.ecr.aws/bottlerocket/bottlerocket-sdk-${BUILDSYS_ARCH}:${BUILDSYS_SDK_VERSION}
```
The Bottlerocket SDK contains the compilers and toolchain that we expect to use for building Bottlerocket and its associated software.  The SDK contains images for both x86_64 and aarch64 that target both x86_64 and aarch64.  If you pull `public.ecr.aws/bottlerocket/bottlerocket-sdk-aarch64:v0.22.0` on either architecture, it still has the compilers and toolchain for aarch64.

We don't need to have BUILDSYS_ARCH (or buildx's VM-based cross-building environment) since we have the correct images for both of these already.

```suggestion
BUILDSYS_SDK_IMAGE = public.ecr.aws/bottlerocket/bottlerocket-sdk-${TARGET_ARCH}:${BUILDSYS_SDK_VERSION}
```

samuelkarp reviewed (changes requested):
I don't think we need this change; the SDK already has the ability to build binaries for the correct target architecture without involving buildx.

Have you run into problems using the SDK for that?

gthao313 commented:
Convert it to draft and will make sure if new change meets @samuelkarp 's expectation. 

solution:
The arm image did run but metadata was not correct (expect arm but amd).
I was trying to use buildx as part of a multi-stage build to attach the correct metadata.

gthao313 committed:
use latest version SDK

this change will make brupop use latest version SDK and
fix aarch64 building failure issue.

gthao313 committed:
update Go to 1.16.5

cbgbt reviewed (approved):
LGTM :shipit:

WilboMo reviewed (approved):
LGTM
